File: Java/src/main/java/com/anexsys/recruitment/challenge/Card.java
package com.anexsys.recruitment.challenge;

/**
 * A Card must be in a 'deck' such that games which use more than one deck can identify the specific card.
 */
class Card {
    private final Suit suit;
    private final Rank rank;
    private final String deckId;
    Card(Suit suit, Rank rank, String deckId) {
        this.suit = suit;
        this.rank = rank;
        this.deckId = deckId;
    }
    Suit suit() {
        return this.suit;
    }
    Rank rank() {
        return this.rank;
    }
    String deckId() {
        return this.deckId;
    }
    @Override
    public String toString() {
        return this.rank + " of " + this.suit + " in " + this.deckId;
    }
    @Override
    public boolean equals(Object obj) {
        Card that;
        return obj != null && obj instanceof Card && (that = (Card) obj).suit == this.suit && that.rank == this.rank && this.deckId.equals(that.deckId);
    }
    @Override
    public int hashCode() {
        return (this.suit == null ? 32 : this.suit.hashCode()) ^ (this.rank == null ? 3 : this.rank.hashCode()) * this.deckId.hashCode();
    }
}

File: DotNet/DotNet/Card.cs
using System;

namespace DotNet
{
    public class Card
    {
        public Suit? Suit { get; }
        public Rank? Rank { get; }
        public string DeckId { get; }

        public Card(Suit? suit, Rank? rank, String deckId)
        {
            Suit = suit;
            Rank = rank;
            DeckId = deckId;
        }

        public override string ToString()
        {
            return $"{Rank} of {Suit} in {DeckId}";
        }

        public override bool Equals(object obj)
        {
            Card that;
            return obj != null && obj is Card && (that = (Card)obj).Suit == Suit && that.Rank == Rank && DeckId.Equals(that.DeckId);
        }

        public override int GetHashCode()
        {
            return (Suit == null ? 32 : Suit.GetHashCode()) ^ (Rank == null ? 3 : Rank.GetHashCode()) * DeckId.GetHashCode();
        }
    }
}
